pkg/controllers/mcs/service_export_controller_test.go

```diff
@@ -0,0 +1,1549 @@
+					},
+					Finalizers: []string{util.EndpointSliceControllerFinalizer},
+				},
+			},
+		},
+		{
+			name:        "update existing work for endpointslice",
+			clusterName: "member1",
+			endpointSlice: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "discovery.k8s.io/v1",
+					"kind":       "EndpointSlice",
+					"metadata": map[string]interface{}{
+						"name":      "eps1",
+						"namespace": "default",
+						"labels": map[string]interface{}{
+							discoveryv1.LabelServiceName: "svc1",
+						},
+					},
+				},
+			},
+			existingWork: &workv1alpha1.Work{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      names.GenerateWorkName("EndpointSlice", "eps1", "default"),
+					Namespace: names.GenerateExecutionSpaceName("member1"),
+					Labels: map[string]string{
+						util.ServiceNamespaceLabel:           "default",
+						util.ServiceNameLabel:                "svc1",
+						util.PropagationInstruction:          util.PropagationInstructionSuppressed,
+						util.EndpointSliceWorkManagedByLabel: "OtherController",
+					},
+				},
+			},
+			expectedWork: &workv1alpha1.Work{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      names.GenerateWorkName("EndpointSlice", "eps1", "default"),
+					Namespace: names.GenerateExecutionSpaceName("member1"),
+					Labels: map[string]string{
+						util.ServiceNamespaceLabel:           "default",
+						util.ServiceNameLabel:                "svc1",
+						util.PropagationInstruction:          util.PropagationInstructionSuppressed,
+						util.EndpointSliceWorkManagedByLabel: "OtherController.ServiceExport",
+					},
+					Finalizers: []string{util.EndpointSliceControllerFinalizer},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := runtime.NewScheme()
+			require.NoError(t, workv1alpha1.Install(s))
+			require.NoError(t, discoveryv1.AddToScheme(s))
+
+			fakeClient := fake.NewClientBuilder().WithScheme(s).Build()
+
+			if tt.existingWork != nil {
+				err := fakeClient.Create(context.TODO(), tt.existingWork)
+				require.NoError(t, err)
+			}
+
+			err := reportEndpointSlice(context.TODO(), fakeClient, tt.endpointSlice, tt.clusterName)
+
+			if tt.expectedError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+
+				workName := names.GenerateWorkName(tt.endpointSlice.GetKind(), tt.endpointSlice.GetName(), tt.endpointSlice.GetNamespace())
+				work := &workv1alpha1.Work{}
+				err = fakeClient.Get(context.TODO(), types.NamespacedName{Name: workName, Namespace: names.GenerateExecutionSpaceName(tt.clusterName)}, work)
+				require.NoError(t, err)
+
+				assert.Equal(t, tt.expectedWork.Name, work.Name)
+				assert.Equal(t, tt.expectedWork.Namespace, work.Namespace)
+
+				// Compare sorted labels
+				expectedLabels := sortedLabels(tt.expectedWork.Labels)
+				actualLabels := sortedLabels(work.Labels)
+				assert.Equal(t, expectedLabels, actualLabels, "Labels do not match")
```

Comment:
A test assert failed:
```log
Expected :endpointslice.karmada.io/managed-by=OtherController.ServiceExport,endpointslice.karmada.io/name=svc1,endpointslice.karmada.io/namespace=default,propagation.karmada.io/instruction=suppressed
Actual   :endpointslice.karmada.io/managed-by=ServiceExport.OtherController,endpointslice.karmada.io/name=svc1,endpointslice.karmada.io/namespace=default,propagation.karmada.io/instruction=suppressed
```

Reply:
I sorted the order of labels and it was working fine locally. I think for _managed-by_ label, i need to sort both the value and the key. What do you think?

Reply:
Maybe it's necessary to eliminate randomness in the test.
ps, I'm also testing locally.

Reply:
Thanks! I'll try a different approach.